### PR TITLE
support python 3.12

### PR DESCRIPTION
**Pull request by TomerBin**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "async-firebase"
-version = "3.2.0"
+version = "3.2.1"
```
Comment (akalex) [inline pyproject.toml:3]: First of all, thank you for contributing to the project. 
As this is not a bugfix, could you please increment the MINOR digit?

**Review by akalex — CHANGES_REQUESTED** [review]

**Comment by akalex** [comment]
I believe you have to use `actions/setup-python@v4` to enable Python 3.12 support.

**Comment by akalex** [comment]
Okay, let's merge this one, and then I will fix CI/CD so as not to bother you with that stuff. @TomerBin, Are you good with that?

**Comment by hxtomer** [comment]
> Okay, let's merge this one, and then I will fix CI/CD to avoid bothering you with that stuff. @TomerBin, Are you good with that?

I've added `allow-prereleases: true` to the CI, maybe it'll work? I prefer to let the unit test run in the CI since i haven't run them locally.

**Comment by akalex** [comment]
> > Okay, let's merge this one, and then I will fix CI/CD to avoid bothering you with that stuff. @TomerBin, Are you good with that?
> 
> I've added `allow-prereleases: true` to the CI, maybe it'll work? I prefer to let the unit test run in the CI since i haven't run them locally.

Okay, that sounds reasonable.

**Comment by TomerBin** [comment]
@akalex Is it possible to grant me permissions to run the 3.12 action so I can test it effectively? Thank you

**Comment by akalex** [comment]
> @akalex Is it possible to grant me permissions to run the 3.12 action so I can test it effectively? Thank you

Hey @TomerBin, sorry for the delay. I adjusted the configuration. You should now be able to Run GitHub actions.

**Review by akalex — APPROVED** [review]
LGTM

**Pull request merged**